alerting/src/main/kotlin/org/opensearch/alerting/alerts/AlertMover.kt

```diff
@@ -116,3 +127,108 @@ private fun alertContentParser(bytesReference: BytesReference): XContentParser {
     XContentParserUtils.ensureExpectedToken(XContentParser.Token.START_OBJECT, xcp.nextToken(), xcp)
     return xcp
 }
+
+/**
+ * Moves defunct active alerts to the alert history index when the corresponding workflow or trigger is deleted.
+ *
+ * The logic for moving alerts consists of:
+ * 1. Find active alerts:
+ *      a. matching workflowId if no workflow is provided (postDelete)
+ *      b. matching workflowid and chained alert triggerIds if monitor is provided (postIndex)
+ * 2. Move alerts over to DataSources.alertsHistoryIndex as DELETED
+ * 3. Delete alerts from monitor's DataSources.alertsIndex
+ * 4. Schedule a retry if there were any failures
+ */
+suspend fun moveAlerts(client: Client, workflowId: String, workflow: Workflow?, monitorCtx: MonitorRunnerExecutionContext) {
+    var alertIndex = ALERT_INDEX
+    var alertHistoryIndex = ALERT_HISTORY_WRITE_INDEX
+    if (workflow != null) {
+        if (
+            workflow.inputs.isNotEmpty() && workflow.inputs[0] is CompositeInput &&
+            (workflow.inputs[0] as CompositeInput).sequence.delegates.isNotEmpty()
+        ) {
+            val i = 0
+            val delegates = (workflow.inputs[i] as CompositeInput).sequence.delegates
+            try {
+                var getResponse: GetResponse? = null
+                while (i < delegates.size && (getResponse == null || getResponse.isExists == false)) {
+                    getResponse =
+                        client.suspendUntil { client.get(GetRequest(ScheduledJob.SCHEDULED_JOBS_INDEX, delegates[0].monitorId), it) }
+                    if (getResponse!!.isExists) {
+                        val monitor =
+                            ScheduledJobUtils.parseMonitorFromScheduledJobDocSource(monitorCtx.xContentRegistry!!, response = getResponse)
+                        if (monitorCtx.alertIndices!!.isAlertInitialized(monitor.dataSources) == false)
```

Comment:
We need to check this at the top level or revert the check in postDelete for `MonitorRunnerService.kt`

Reply:
we need to first get a delegate monitor in workflow to get it's datasources to check the alert indices being used - if they are custom or not.

only then we can check if they are initialized